contrib/autorandr_dbus_monitor.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+# Copyright 2021 Christophe-Marie Duquesne
+
+# Check if dbus-monitor is installed
+command -v dbus-monitor >/dev/null 2>&1 || \
+    { echo >&2 "dbus-monitor is required but is not installed. Exiting"; exit 1; }
+
+pipe=$(mktemp -u /tmp/autorandr.XXXXXXXX)
+
+trap "rm -f $pipe" EXIT
+
+if [[ ! -p $pipe ]]; then
+    mkfifo $pipe
+fi
+
+suspend_event="type='signal',interface='org.freedesktop.login1.Manager',member='PrepareForSleep'"
+lid_event="type='signal',path=/org/freedesktop/UPower,member=PropertiesChanged"
+display_event="type='signal',path=/org/freedesktop/ColorManager"
```

Comment:
Note that this `ColorManager` dbus signal comes from `colord`, which may not be configured/running for barebone WM-only setups.
See: https://github.com/hughsie/colord/blob/master/src/org.freedesktop.ColorManager.xml

Reading https://www.freedesktop.org/software/colord/intro.html, it says that colord can send events for multiple subsystems, including xrandr (screen changes).
This means that with that trigger as is, you'd run autorandr when you plug a webcam or a printer, and I don't think you want that ^^

Reply:
This is the only way I found to detect newly plugged monitors via dbus. If this triggers when a printer is plugged, I don't think it is a big deal: better trigger too often than not often enough.

Anyway, I have been using this script for 3 years with under i3-wm and it works flawlessly for me.